### PR TITLE
feat(InputGroup,TextInput): add tokens

### LINKED ISSUE #6191: Tokens for Input group, number input

### PR DESCRIPTION
**Pull request by srambach**
Fixes #6191 

Also adjusts a padding on the form element because it was too close to the status icon.
I can combine the borders into one variable if desired.

Convenience links to some things affected:
https://patternfly-pr-6201.surge.sh/components/number-input
https://patternfly-pr-6201.surge.sh/components/input-group
https://patternfly-pr-6201.surge.sh/components/text-input-group

### CODE REVIEW

File: src/patternfly/components/InputGroup/input-group.scss [hunk 1]
```diff
@@ -1,40 +1,41 @@
...
+  --#{$input-group}__item--m-box--PaddingRight: var(--pf-t--global--spacer--sm);
+  --#{$input-group}__item--m-box--PaddingLeft: var(--pf-t--global--spacer--sm);
+  --#{$input-group}__item--m-box--BackgroundColor: var(--pf-t--global--background--color--secondary--default);
+  --#{$input-group}__item--m-box--BorderWidth: var(--#{$input-group}__item--BorderWidth--base);
+  --#{$input-group}__item--m-box--BorderRadius: var(--pf-t--global--border--radius--small);
   --#{$input-group}__item--m-box--BorderTopColor: var(--#{$input-group}__item--BorderColor--base);
   --#{$input-group}__item--m-box--BorderRightColor: var(--#{$input-group}__item--BorderColor--base);
-  --#{$input-group}__item--m-box--BorderBottomColor: var(--#{$input-group}__item--BorderColor--accent);
+  --#{$input-group}__item--m-box--BorderBottomColor: var(--#{$input-group}__item--BorderColor--base);
   --#{$input-group}__item--m-box--BorderLeftColor: var(--#{$input-group}__item--BorderColor--base);
```
Comment (mattnolting) [inline src/patternfly/components/InputGroup/input-group.scss:22]: Looks like this is broken. Should we remove the variant?

Screenshot 2024-01-08 at 4 38 31 PM

**Review by mattnolting — COMMENTED** [review]
Looks great! One question about the `pf-m-box` variant.

File: src/patternfly/components/InputGroup/input-group.scss (hunk 1 above)
Comment (srambach) [reply src/patternfly/components/InputGroup/input-group.scss:22]: That's used to put the extra things around it, like the $ or "@example.com" 
I think we still want that?

**Comment by lboehling** [comment]
LGTM!! thank you @srambach 

I did notice some weird things happening in the toolbar demos, but i'm guessing that's because the examples aren't using input group?  should they be? 
https://patternfly-pr-6201.surge.sh/components/toolbar/html-demos/toolbar-with-validation-on-desktop/
https://patternfly-pr-6201.surge.sh/components/toolbar/html-demos#toolbar-attribute-value-single-select-filter-on-desktop

**Review by mattnolting — APPROVED** [review]
WooHoo! LPTM, great work 🎉

**Review by mcoker — APPROVED** [review]
🎃👍

**Pull request merged**